cog/daily_charge.py

```
@@ -23,95 +26,242 @@
-        await interaction.response.send_message(embed=self.embed, ephemeral=True)
+        # 紀錄log
+        # TODO: record both executed time and datetime.now() after logger is implemented
+        # pylint: disable-next = line-too-long
+        print(f"{user.id},{user} Get 5 point by daily_charge {datetime.now()}")
 
-    @discord.slash_command(name="charge", description="每日充電")
-    async def charge(self, interaction):
-        user_id = interaction.user.id
-        connection, cursor = link_sql()  # SQL 會話
+    # TODO: inherit a MySQLCursorAbstract to add method about these or consider to add self.cursor
+    def get_last_charged(self, user: discord.User | discord.Member, cursor):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_last_charged`

(ANN201)